plugins/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,527 @@
+        sample: user-test-123
+    attributes:
+        description: Attributes applied to this user.
+        type: dict
+        returned: always
+        sample:
+            attr1: ["val1", "val2", "val3"]
+
+'''
+
+from ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak import KeycloakAPI, camel, \
+    keycloak_argument_spec, get_token, KeycloakError
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+
+    meta_args = dict(
+        # Remove alias `username` on `auth_username`
+        auth_username=dict(type='str'),
+
+        state=dict(default='present', choices=['present', 'absent']),
+        realm=dict(default='master'),
+
+        username=dict(type='str'),
+
+        access=dict(type='dict'),
+        attributes=dict(type='dict'),
+        client_consents=dict(type='list', elements='dict',
+                             options=dict(
+                                 # UserConsentRepresentation
+                                 client_id=dict(
+                                     type='str', aliases=['clientId']),
+                                 granted_client_scopes=dict(
+                                     type='list', elements='str', aliases=['grantedClientScopes']),
+                             ),
+                             aliases=['clientConsents']),
+        credentials=dict(type='list', elements='dict',
+                         options=dict(
+                             # CredentialRepresentation
+                             priority=dict(type='int'),
+                             type=dict(type='str'),
+                             temporary=dict(type='bool'),
+                             value=dict(type='str'),
+                         )),
+        disableable_credential_types=dict(type='list', elements='str', aliases=[
+                                          'disableableCredentialTypes']),
+        email=dict(type='str'),
+        email_verified=dict(type='bool', aliases=['emailVerified']),
+        enabled=dict(type='bool'),
+        first_name=dict(type='str', aliases=['firstName']),
+        groups=dict(type='list', elements='str'),
+        id=dict(type='str'),
+        last_name=dict(type='str', aliases=['lastName']),
+        not_before=dict(type='int', aliases=['notBefore']),
+        origin=dict(type='str'),
+        realm_roles=dict(type='list', elements='str', aliases=['realmRoles']),
+        required_actions=dict(type='list', elements='str',
+                              aliases=['requiredActions']),
+        service_account_client_id=dict(
+            type='str', aliases=['serviceAccountClientId']),
+    )
+
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           required_one_of=([['id', 'username'],
```

Comment:
Would it be beneficial to use `mutually_exclusive` i.e. to allow `username`, or `id` but not both? - See https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec

Reply:
I agree with you, for sure. The only reason why I choose to have similar solutions from other files instead of **best solution**, is that I think keeping consistency for those files is more important in the first place, then we can think about to refactoring later.

Reply:
Adding such restrictions to existing modules is a breaking change and needs a long deperecation period (and extra code to inform users about it), that's usually the main reason for it not being done. But for new modules we should better stick to the best solution :)

Reply:
This is a brand new module (`keycloak_user.py` didn't exist prior to this PR), so I didn't think the concern of deprecation applied, (i.e we should instead try to get it as right as possible up front)